src/ucs/async/async.c

```diff
@@ -461,16 +458,16 @@ ucs_async_alloc_handler(int min_id, int max_id, ucs_async_mode_t mode,
         goto err_free;
     }
 
+    if (async != NULL) {
+        ucs_atomic_fadd32(&async->num_handlers, 1);
```

Comment:
can we delete the field async->num_handlers?

Reply:
does it mean removing this if block https://github.com/openucx/ucx/blob/313ba9f7fded9688096660245ae1c574ab6bdd73/src/ucs/async/async.c#L395?

Reply:
yes

Reply:
done